hw2/01-coins.js

```diff
@@ -1,7 +1,46 @@
-/** Exercise 01 - Coins **/
+/* Exercise 01 - Coins */
 
+// eslint-disable-next-line consistent-return
 const calculateChange = (input) => {
-  // Add your code here
+  if (input <= 0) return "input should be positive";
+  if (input > 10) return "Error: the number is too large";
+
+  // eslint-disable-next-line prefer-const
+  let [dollars, remaining] = input
+    .toString()
+    .split(".")
+    .map((i) => parseInt(i, 10));
+
+  let answer = "";
+
+  const quarters = Math.floor(remaining / 25);
+  remaining -= quarters * 25;
+
+  const dime = Math.floor(remaining / 10);
+  remaining -= dime * 10;
+
+  const nickel = Math.floor(remaining / 5);
+  remaining -= nickel * 5;
+
+  const pennies = Math.floor(remaining);
+
+  if (dollars) {
+    answer = `${answer} ${dollars} dollar${dollars > 2 ? "s" : ""},`;
+  }
+  if (quarters) {
+    answer = `${answer} ${quarters} quarter${quarters >= 2 ? "s" : ""},`;
+  }
+  if (dime) {
+    answer = `${answer} ${dime} dime${dime >= 2 ? "s" : ""},`;
+  }
+  if (nickel) {
+    answer = `${answer} ${nickel} nickel${nickel >= 2 ? "s" : ""},`;
+  }
+  if (pennies) {
+    answer = `${answer} ${pennies} penn${pennies >= 2 ? "ies" : "y"}`;
+  }
+
+  console.log(answer);
```

Comment:
1. It seems you have not returned any value, which would end up showing undefined after each function call. Maybe you can update that.
2. If possible update the printing statement as suggested by Caterina(Not a major requirement)
Expected : $4.62 ==> 4 dollars, 2 quarters, 1 dime, 2 pennies
Actual : 4 dollars, 2 quarters, 1 dime, 2 pennies

Reply:
Not sure what happened here, but i've already made these changes.

Reply:
Got it. I committed them yesterday but forgot to push. lol
Thanks for suggestions though!